builtin/logical/pki/path_roles.go

```diff
@@ -727,6 +727,9 @@ func (b *backend) pathRoleCreate(ctx context.Context, req *logical.Request, data
 		}
 	} else {
 		*entry.GenerateLease = data.Get("generate_lease").(bool)
+		if *entry.GenerateLease {
+			warning = "it is strongly encouraged to disable generate_lease and rely on PKI's native capabilities when possible; this option causes issues with large numbers of issued certificates"
+		}
```

Comment:
Maybe too strong since there are legit use cases?  Or at least ammend to "this option may cause issues..."

Reply:
Have we collected what those are? At a large enough scale even for legitimate use cases, they'd still run into the same issues with this parameter...

I think last I looked, agent will do renewal of cert secrets without a backing lease just fine. Maybe it doesn't check revocation correctly though and should be updated to fix that?

Maybe just remove "strongly"?

Reply:
@sgmiller Updated, let me know what you think?

Reply:
Still want the softening "may" or "can" cause in there.

Reply:
Ok, updated, thanks!